modules/motions/hooks/usePeriodLimitsInfo.ts

```diff
@@ -0,0 +1,192 @@
+import moment from 'moment'
+import { LimitCheckerAbi, EasyTrackAbi } from 'generated'
+import { createContractHelpers } from 'modules/blockChain/utils/createContractHelpers'
+import {
+  ContractLegoDAIRegistry,
+  ContractEasyTrack,
+  ContractAllowedRecipientRegistry,
+  ContractLegoLDORegistry,
+  ContractRccDAIRegistry,
+  ContractAtcDAIRegistry,
+  ContractGasFunderETHRegistry,
+  ContractPmlDAIRegistry,
+} from 'modules/blockChain/contracts'
+import { useWeb3 } from 'modules/blockChain/hooks/useWeb3'
+import { useSWR, SWRResponse } from 'modules/network/hooks/useSwr'
+import {
+  getLimits,
+  LimitsType,
+  getPeriodData,
+  PeriodDataType,
+} from 'modules/motions/utils'
+import { MotionType } from 'modules/motions/types'
+import { EvmUnrecognized } from 'modules/motions/evmAddresses'
+
+type ContractLimitsMethods = {
+  getLimitParameters: LimitCheckerAbi['getLimitParameters']
+  getPeriodState: LimitCheckerAbi['getPeriodState']
+}
+
+type UsePeriodLimitInfoProps<T> = {
+  address: string
+  contract: T
+  swrKey: string
+  isPending?: boolean
+}
+
+export type usePeriodLimitsInfoResultData = {
+  limits: LimitsType
+  periodData: PeriodDataType
+  motionDuration: number
+  isEndInNextPeriod: boolean
+}
+
+type UsePeriodLimitInfo = <T extends ContractLimitsMethods>(
+  data: UsePeriodLimitInfoProps<T>,
+) => SWRResponse<{
+  limits: LimitsType
+  periodData: PeriodDataType
+  motionDuration: number
+  isEndInNextPeriod: boolean
+}>
```

Comment:
Seems like duplication of `usePeriodLimitsInfoResultData` type, declared before. Maybe use it here?